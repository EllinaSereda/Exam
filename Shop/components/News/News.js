import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import './News.css';


class News extends React.PureComponent {

  static propTypes = {
    news:PropTypes.object,
  };
  

  createText=(text)=>{

    text.length>200?text=text.substr(0,200):text;
    var code=text.slice(0);
    code=code.split(/<.*?br.*?>/ig).map ((v,i,a) =>{
      if (i==a.length-1)
      v=v+'...'
      return v;
    });
    for(let i=1,j=0;i<code.length;i=i+2, j++){
      code.splice(i,0,<br key={j}/>);
    }

    
    return code;
  }



  render() {
      console.log(this.props.news);
    return <div className='News' key={this.props.news.code}>
    <div className="Name"><span>{this.props.news.name}</span></div>
    <NavLink to={"/news/"+this.props.news.code} exact className="PageLink" activeClassName="ActivePageLink">
    <img src={this.props.news.url}/></NavLink> 
    <p>{this.createText(this.props.news.info)}</p>
    <NavLink to={"/news/"+this.props.news.code} exact className="PageLink" activeClassName="ActivePageLink"><input type="button" value="Читать"/></NavLink>
    </div>

  }

}


export default News;
