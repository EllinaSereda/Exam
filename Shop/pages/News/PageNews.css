.PageNews{
    width:80%;
    margin:auto;
    margin-bottom: 20px;
    
}
.News{
    box-shadow: 0 0 20px rgba(0,0,0,0.3);
    padding-top:30px;
    padding-bottom:90px;
}
.PageNews img{
    width:90%;
    display:block;
    margin:20px auto;
}
.PageNews .Name{
    text-align:center;
    background:url("https://firebasestorage.googleapis.com/v0/b/finalproject-d5d17.appspot.com/o/lineb.png?alt=media&token=63c883e8-8570-4e54-b093-4e2ea3626ba8")  no-repeat,
    url("https://firebasestorage.googleapis.com/v0/b/finalproject-d5d17.appspot.com/o/liner.png?alt=media&token=a19b78ba-62f2-41b4-b762-61c99adcf478")  no-repeat ;
    background-position: center left, center right;
    padding:10px;
}
.PageNews .Name span{
    background-color: white;
    padding:5px;
    font-size:18px;
    font-weight:bold;
}
.PageNews p{
    margin:auto;
    width:90%;
}
 .PageNews input[type="button"]{
     float:right;
     width:150px;
    font-family: 'Montserrat Alternates', sans-serif;
    background-color:rgba(156, 60, 118, 0.7);
    color:white;
    border-radius:10px;
    font-size:18px;
    padding:10px;
    margin:10px 30px;
 }
 .PageNews input[type="button"]:after{
    content:"";
    clear:both;
 }
 .PageNews input[type="button"]:hover{
    background-color:rgba(156, 60, 118, 1);
}
.PageNews h2{
    text-transform: uppercase;
    font-size: 22px;
    font-weight:bold;
    color:#631B42;
    padding:30px 0;
}
