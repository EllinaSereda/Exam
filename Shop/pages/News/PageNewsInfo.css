.NewsInfo{
    box-shadow: 0 0 20px rgba(0,0,0,0.3);
    padding:30px 0;
    margin-bottom:30px;
}
.PageNewsInfo{
    width:80%;
    margin: auto;
    margin-bottom: 20px;
    
    
}
.PageNewsInfo img{
    width:90%;
    display:block;
    margin:20px auto;
}
.PageNewsInfo p{
    padding:10px 30px;
}
.PageNewsInfo p::first-letter{
    font-size:24px;
}
.PageNewsInfo .Name{
    text-align:center;
    background:url("https://firebasestorage.googleapis.com/v0/b/finalproject-d5d17.appspot.com/o/lineb.png?alt=media&token=63c883e8-8570-4e54-b093-4e2ea3626ba8")  no-repeat,
    url("https://firebasestorage.googleapis.com/v0/b/finalproject-d5d17.appspot.com/o/liner.png?alt=media&token=a19b78ba-62f2-41b4-b762-61c99adcf478")  no-repeat ;
    background-position: center left, center right;
    padding:10px;
}
.PageNewsInfo .Name span{
    background-color: white;
    padding:5px;
    font-size:18px;
    font-weight:bold;
}
.PageNewsInfo h2{
    text-transform: uppercase;
    font-size: 22px;
    font-weight:bold;
    color:#631B42;
    padding:30px 0;
}