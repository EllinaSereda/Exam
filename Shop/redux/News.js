const NEWS_CREATE='NEWS_CREATE'; //записываем информацию "Новости" полученную из базы данных

const news_create=function(info) {
  return {
    type: NEWS_CREATE,
    info: info,
  };
}

export {
  news_create,NEWS_CREATE,
}
