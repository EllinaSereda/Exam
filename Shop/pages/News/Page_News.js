import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import News from '../../components/News/News';
import './PageNews.css';


class Page_News extends React.PureComponent {

  static propTypes = {
    news:PropTypes.shape({
      news:PropTypes.array,
    }),
  };




  render() {
    return <div className="PageNews">
    <h2>Новости</h2>
     {this.props.news.news.map(v=><News news={v}/>)}
    </div> ;

  }

}

const mapStateToProps = function (state) {
  return {
    news: state.news,
  };
};

export default connect(mapStateToProps)(Page_News);
