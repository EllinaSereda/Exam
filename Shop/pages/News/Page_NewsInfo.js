import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import './PageNewsInfo.css';

class Page_NewsInfo extends React.PureComponent {

  static propTypes = {
    new:PropTypes.shape({
      news:PropTypes.array,
    }),
  }

  state={
  }
  createText=(info,id)=>{
    let el=info.filter(v=>v.code==id)[0];
    let text=el.info.split(/<.*?br.*?>/ig).map ((v,i,a) =>v);
    let code=[];
    for (let i=0;i<text.length;i++){
      code[i]=<p>{text[i]}</p>;
    }
    let codeNews=<div className='NewsInfo' key={id}>
    <div class="Name"><span>{el.name}</span></div>
    <img src={el.url}/>
    {code}
    </div>
    return codeNews;
  }

  render() {
    const id = this.props.match.params.id;
    console.log('Page_NewsInfo render');
    console.log((this.props.new.news));
    return <div className="PageNewsInfo">
    <h2>Новости</h2>
      {this.createText(this.props.new.news,id)}
    </div> ;

  }

}

const mapStateToProps = function (state) {
  return {
    new: state.news,
  };
};

export default connect(mapStateToProps)(Page_NewsInfo);

//