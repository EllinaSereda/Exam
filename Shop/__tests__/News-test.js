"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import News from '../components/News/News';
import { BrowserRouter } from 'react-router-dom';

test('работа News', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
   <BrowserRouter><div><News news={{code:0,info:'новость',name:'новость1',url:'#'}}/></div></BrowserRouter>
  );


  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();


});
