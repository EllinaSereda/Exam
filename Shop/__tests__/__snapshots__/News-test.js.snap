// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа News 1`] = `
<div>
  <div
    className="News"
  >
    <div
      className="Name"
    >
      <span>
        новость1
      </span>
    </div>
    <a
      aria-current={null}
      className="PageLink"
      href="/news/0"
      onClick={[Function]}
      style={undefined}
    >
      <img
        src="#"
      />
    </a>
    <p>
      новость...
    </p>
    <a
      aria-current={null}
      className="PageLink"
      href="/news/0"
      onClick={[Function]}
      style={undefined}
    >
      <input
        type="button"
        value="Читать"
      />
    </a>
  </div>
</div>
`;
